backend/src/services/sessionService.js

```diff
@@ -1,7 +1,21 @@
 const jwt = require('jsonwebtoken');
 
+
 module.exports = {
   createToken: data => {
     return jwt.sign(data, process.env.ACCESS_TOKEN_SECRET || 'testing');
   },
+
+   authenticate : (req, res, next) => {
+    const token = req.header('auth_token');
+    if (!token) return res.sendStatus(401).send('Access Denied');
+  
+    try {
+      const verified = jwt.verify(token, process.env.ACCESS_TOKEN_SECRET);
+      req.user = verified;
+      next();
+    } catch (err) {
+      res.sendStatus(400).send('Invalid Token');
+    }
+  }
```

Comment:
@LLaszlo89 this is a middleware, should be in the middleware folder as an authentication middleware, but basically line 14 should be here as a verifyToken method, that you can import there